File: PhaseJumpPro/Assets/phasejumppro/Types/SomeReference.cs
using System;

/*
 * RATING: 5 stars
 * Simple utility
 * CODE REVIEW: 11/1/22
 */
namespace PJ
{
    /// <summary>
    /// Interface for a reference type. Allows either strong or weak reference to value
    /// `ReferenceType`, not named `Reference, to avoid name collision with WeakReference
    /// </summary>
    /// <typeparam name="T"></typeparam>
    public interface SomeReference<T>
    {
        public T Value { get; set; }
    }
}

File: Experimental/Swift/Old Swift/Core/SomeReference.swift
import Foundation

/*
 RATING: 5 stars
 Simple wrapper
 CODE REVIEW: 4/11/22
 */

/// Allows us to abstract a weak or strong reference based on use
/// Example: cyclic vs acyclic graph
public protocol SomeReference {
    associatedtype Value

    var value: Value? { get }
}
